src/lib/teads-curve/index.ts

```diff
@@ -0,0 +1,134 @@
+import {IImpactModelInterface} from '../interfaces';
+import Spline from 'typescript-cubic-spline';
+import {KeyValuePair} from '../../types/boavizta';
+
+export class TeadsCurveModel implements IImpactModelInterface {
+  // Defined for compatibility. Not used in TEADS.
+  authParams: object | undefined;
+  // name of the data source
+  name: string | undefined;
+  // tdp of the chip being measured
+  tdp = 100;
+  // default power curve provided by the Teads Team
+  curve: number[] = [0.12, 0.32, 0.75, 1.02];
+  // default percentage points
+  points: number[] = [0, 10, 50, 100];
+  // spline interpolation of the power curve
+  spline: Spline = new Spline(this.points, this.curve);
+
+  /**
+   * Defined for compatibility. Not used in TEADS.
+   */
+  authenticate(authParams: object): void {
+    this.authParams = authParams;
+  }
+
+  /**
+   *  Configures the TEADS Plugin for IEF
+   *  @param {string} name name of the resource
+   *  @param {Object} staticParams static parameters for the resource
+   *  @param {number} staticParams.tdp Thermal Design Power in Watts
+   */
+  async configure(
+    name: string,
+    staticParams: object | undefined = undefined
+  ): Promise<IImpactModelInterface> {
+    this.name = name;
+
+    if (staticParams === undefined) {
+      throw new Error('Required Parameters not provided');
+    }
+
+    if ('tdp' in staticParams) {
+      this.tdp = staticParams?.tdp as number;
+    } else {
+      throw new Error(
+        '`tdp` Thermal Design Power not provided. Can not compute energy.'
+      );
+    }
+
+    if ('curve' in staticParams) {
+      this.curve = staticParams?.curve as number[];
+      this.spline = new Spline(this.points, this.curve);
```

Comment:
Similar point to the one I made on TEADs AWS. Can we have this configurable
interpolation: linear | curve (default is linear since that's how everyone is using this these days).
Also are you making the curve itself configurable?

Reply:
I am making the curve itself configurable. So giving in a points array of [0,100] and curve array of [0.1,1.05] for example in static params will make the computation in a linear manner. 

Linear interpolation can only performed between two points. I might need to do another approximation if we want to do multiple points and it is certainly not the appropriate one.